librz/core/cmd/cmd_open.c

```diff
@@ -1251,3 +778,322 @@ RZ_IPI RzCmdStatus rz_open_binary_reload_handler(RzCore *core, int argc, const c
+	return RZ_CMD_STATUS_OK;
+}
+
+static RzCmdStatus open_core_file(RzCore *core, const char *filename) {
+	if (core->tasks.current_task != core->tasks.main_task) {
+		RZ_LOG_ERROR("This command can only be executed on the main task!\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	rz_core_task_sync_end(&core->tasks);
+	rz_core_fini(core);
+	rz_core_init(core);
+	rz_core_task_sync_begin(&core->tasks);
+	if (!rz_core_file_open(core, filename, RZ_PERM_R, 0)) {
+		RZ_LOG_ERROR("Cannot open file '%s'\n", filename);
+		return RZ_CMD_STATUS_ERROR;
+	}
+	ut64 baddr = rz_config_get_i(core->config, "bin.baddr");
+	return bool2status(rz_core_bin_load(core, NULL, baddr));
+}
+
+RZ_IPI RzCmdStatus rz_open_core_file_handler(RzCore *core, int argc, const char **argv) {
+	return open_core_file(core, argv[1]);
+}
+
+RZ_IPI RzCmdStatus rz_open_malloc_handler(RzCore *core, int argc, const char **argv) {
+	int len = (int)rz_num_math(core->num, argv[1]);
+	if (len < 0) {
+		RZ_LOG_ERROR("Invalid length %d.\n", len);
+		return RZ_CMD_STATUS_ERROR;
+	}
+
+	RzCmdStatus res = RZ_CMD_STATUS_ERROR;
+	ut8 *data = RZ_NEWS(ut8, len);
+	if (!data) {
+		return RZ_CMD_STATUS_ERROR;
+	}
+	rz_io_read_at(core->io, core->offset, data, len);
+
+	char uri[100];
+	rz_strf(uri, "malloc://%d", len);
+	RzCoreFile *cfile = rz_core_file_open(core, uri, RZ_PERM_RWX, 0);
+	if (!cfile) {
+		RZ_LOG_ERROR("Cannot open '%s'.\n", uri);
+		goto err;
+	}
+
+	rz_core_bin_load(core, uri, 0);
+
+	RzIODesc *desc = rz_io_desc_get(core->io, cfile->fd);
+	rz_warn_if_fail(desc);
+	rz_io_desc_write_at(desc, 0, data, len);
+	res = RZ_CMD_STATUS_OK;
+
+err:
+	free(data);
+	return res;
+}
+
+static RzCmdStatus open_nobin_file(RzCore *core, const char *uri, ut64 addr, int perms) {
+	if (!strcmp(uri, "=")) {
+		uri = "malloc://512";
+	}
+
+	RzIODesc *desc = rz_io_open_at(core->io, uri, perms, 0644, addr, NULL);
+	if (!desc || desc->fd == -1) {
+		RZ_LOG_ERROR("Cannot open '%s' at %" PFMT64x ".\n", uri, addr);
+		return RZ_CMD_STATUS_ERROR;
+	}
+
+	core->num->value = desc->fd;
+	rz_core_block_read(core);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_open_nobin_handler(RzCore *core, int argc, const char **argv) {
+	ut64 addr = argc > 2 ? rz_num_math(core->num, argv[2]) : 0;
+	int perms = argc > 3 ? rz_str_rwx(argv[3]) : RZ_PERM_R;
+	return open_nobin_file(core, argv[1], addr, perms);
+}
+
+RZ_IPI RzCmdStatus rz_open_nobin_write_handler(RzCore *core, int argc, const char **argv) {
+	ut64 addr = argc > 2 ? rz_num_math(core->num, argv[2]) : 0;
+	int perms = argc > 3 ? rz_str_rwx(argv[3]) : RZ_PERM_RW;
+	return open_nobin_file(core, argv[1], addr, perms);
+}
+
+RZ_IPI RzCmdStatus rz_reopen_handler(RzCore *core, int argc, const char **argv) {
+	int fd;
+	if (argc > 1) {
+		fd = (int)rz_num_math(NULL, argv[1]);
+	} else {
+		if (!core->io || !core->io->desc) {
+			RZ_LOG_ERROR("Cannot find current file.\n");
+			return RZ_CMD_STATUS_ERROR;
+		}
+		fd = core->io->desc->fd;
+	}
+	rz_core_io_file_open(core, fd);
```

Comment:
Check the return value please

Reply:
`RZ_IPI void rz_core_io_file_open(RzCore *core, int fd)`.